packages/tools/devtools/devtools-core/src/test/DefaultVisualizers.spec.ts

```diff
@@ -1730,6 +1730,380 @@ describe("DefaultVisualizers unit tests", () => {
 		expect(result).to.deep.equal(expected);
 	});
 
+	it.only("SharedTree: Renders multiple allowed types in SharedTree's root field", async () => {
+		const factory = SharedTree.getFactory();
+		const builder = new SchemaFactory("shared-tree-test");
+
+		const sharedTree = factory.create(
+			new MockFluidDataStoreRuntime({ idCompressor: createIdCompressor() }),
+			"test",
+		);
+
+		const view = sharedTree.viewWith(
+			new TreeViewConfiguration({ schema: [builder.string, builder.number] }),
+		);
+		view.initialize(23);
+
+		const result = await visualizeSharedTree(
+			sharedTree as unknown as ISharedObject,
+			visualizeChildData,
+		);
+
+		const expected = {
+			children: {
+				root: {
+					value: 23,
+					nodeKind: "ValueNode",
+					tooltipContents: {
+						schema: {
+							nodeKind: "TreeNode",
+							children: {
+								name: {
+									nodeKind: "ValueNode",
+									value: "com.fluidframework.leaf.number",
+								},
+							},
+						},
+					},
+				},
+			},
+			nodeKind: "FluidTreeNode",
+			tooltipContents: {
+				schema: {
+					nodeKind: "TreeNode",
+					children: {
+						name: {
+							nodeKind: "ValueNode",
+							value: "root",
+						},
+						allowedTypes: {
+							value: "com.fluidframework.leaf.string | com.fluidframework.leaf.number",
+							nodeKind: "ValueNode",
+						},
+					},
+				},
+			},
+			fluidObjectId: "test",
+			typeMetadata: "SharedTree",
+		};
+
+		expect(result).to.deep.equal(expected);
+	});
+
+	it.only("SharedTree: Renders multiple allowed types in SharedTree's root field", async () => {
```

Comment:
I don't think this more complex test case is needed, FWIW. The earlier one that uses primitive node kinds should be sufficient (the behavior shouldn't depend on whether or not the allowed kinds of nodes are primitives or not).